Shared/Samples/Set visibility of subtype sublayer/SetVisibilityOfSubtypeSublayerView.Model.swift

```diff
@@ -0,0 +1,148 @@
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+extension SetVisibilityOfSubtypeSublayerView {
+    /// The view model for this sample.
+    @MainActor
+    class Model: ObservableObject {
+        /// A map with a streets basemap style.
+        let map = Map(basemapStyle: .arcGISStreetsNight)
+        
+        /// A feature table for the electrical network in this sample.
+        private let featureTable = ServiceFeatureTable(url: .featureServiceURL)
+        
+        /// A subtype feature layer created from the service feature table.
+        private let subtypeFeatureLayer: SubtypeFeatureLayer
+        
+        /// The subtype sublayer of the subtype feature layer in this sample.
+        private var subtypeSublayer: SubtypeSublayer?
+        
+        /// The renderer of the subtype feature layer.
+        private var originalRenderer: Renderer?
+        
+        /// The  subtype sublayer's label definition.
+        private let labelDefinition: LabelDefinition = {
+            // Make and stylize the text symbol.
+            let textSymbol = TextSymbol(color: .blue, size: 10.5)
+            textSymbol.backgroundColor = .clear
+            textSymbol.outlineColor = .white
+            textSymbol.haloColor = .white
+            textSymbol.haloWidth = 2
+            // Make a label definition and adjust its properties.
+            let labelExpression = SimpleLabelExpression(simpleExpression: "[nominalvoltage]")
+            let labelDefinition = LabelDefinition(labelExpression: labelExpression, textSymbol: textSymbol)
+            labelDefinition.placement = .pointAboveRight
+            labelDefinition.usesCodedValues = true
+            return labelDefinition
+        }()
+        
+        /// The current scale of the map.
+        var currentScale: Double = .zero
+        
+        /// The map's current scale value in text.
+        @Published var currentScaleText: String = ""
+        
+        /// The subtype sublayer's minimum scale value in text.
+        @Published private(set) var minimumScaleText: String = "Not Set"
+        
+        init() {
+            map.initialViewpoint = .initialViewpoint
+            subtypeFeatureLayer = SubtypeFeatureLayer(featureTable: featureTable)
+            subtypeFeatureLayer.scalesSymbols = false
+        }
+        
+        deinit {
+            ArcGISEnvironment.authenticationManager.arcGISCredentialStore.removeAll()
+        }
+        
+        /// Performs important tasks including adding credentials, loading and adding operational layers.
+        func setup() async throws {
+            try await ArcGISEnvironment.authenticationManager.arcGISCredentialStore.add(.publicSample)
+            try await subtypeFeatureLayer.load()
+            map.addOperationalLayer(subtypeFeatureLayer)
+            guard let subtypeSublayer = subtypeFeatureLayer.sublayer(withSubtypeName: "Street Light") else { fatalError("Failed to initialize subtype sublayer") }
```

Comment:
Now that subtypeSublayer is optional, you don't need to crash with a fatalError here. The question is should you return or throw an error?  If you simply return in the case where the sublayer cannot be found then `setup()` basically fails and gives no indication of the failure. That means the code that calls `setup()` won't catch an error and it will think that setup succeeded when it didn't.

What I think should be done is something like this:

```swift
            guard let subtypeSublayer = subtypeFeatureLayer.sublayer(withSubtypeName: "Street Light") else {
                throw SetupError.cannotFindSublayer
            }
```

where you have an error defined something like this:

```swift
extension SetVisibilityOfSubtypeSublayerView.Model {
    enum SetupError: Error {
        case cannotFindSublayer
    }
}
```

You should test that to see what it shows the user in the case of the error here:
```
            .alert(isPresented: $isShowingAlert, presentingError: error)
```

If it doesn't show a nice message the error may need to conform to something like LocalizedError.

Reply:
Here is the presented error message:

<img src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/117859673/67bd0e18-0000-4bfe-b834-de74b080e6b2" width="400">

Reply:
Try conforming to LocalizedError and see if that presents a nice error message.

Reply:
Yes that works

<img src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/117859673/6a2c5ecf-3e9e-42a6-9ea7-d5f4182e73c0" width="400">